src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -138,6 +138,21 @@ internal partial class Binder
                     hasErrors);
             }
 
+            if (conversion.Kind == ConversionKind.ObjectCreation)
+            {
+                var convertedObjectCreation = ConvertObjectCreationExpression(syntax, (BoundUnconvertedObjectCreationExpression)source, isCast, destination, diagnostics);
+                return new BoundConversion(
```

Comment:
We shouldn't do this for a type parameter destination, those would never have arguments and we don't track `WasTargetTyped` for those.

Reply:
We may need to do this for delegates though.
```cs
#nullable enable
var c = Get("");
c = new(() => null); // warning expected
static System.Func<T> Get<T>(T t) => () => t;
```